src/commands/org/login/sfdx-url.ts

```diff
@@ -33,10 +33,18 @@ export default class LoginSfdxUrl extends AuthBaseCommand<AuthFields> {
     'sfdx-url-file': Flags.file({
       char: 'f',
       summary: messages.getMessage('flags.sfdx-url-file.summary'),
-      required: true,
+      required: false,
```

Comment:
We should add this to both `sfdx-url-file` and `sfdx-url-stdin` to ensure that people provide one of those

```
exactlyOne: ['sfdx-url-file', 'sfdx-url-stdin']
```

We should also add `exclusive: ['sfdx-url-stdin']` to make sure that people don't provide both

Reply:
@mdonnalley I added these suggestions but kept the validation logic, otherwise I get a compiler error that `sfdxAuthUrl` could be undefined. I think that's because its not smart enough to know that one of the flags are guaranteed to be provided. Let me know if you know a work-around, but I think that's consistent with other implementations.

Reply:
@k-capehart I think `exclusive` is implied in `exactlyOne` so there's not actually a need for both - you only need `exactlyOne` on both flags

Sorry about that - should have tested it all out before asking you to make a change!

Reply:
@mdonnalley Oh makes sense, that explains the repetition in error messages I was seeing. I removed it 👍